Formula/a/apache-arrow.rb

```diff
@@ -79,10 +78,12 @@ def install
       -DARROW_WITH_UTF8PROC=ON
       -DARROW_INSTALL_NAME_RPATH=OFF
       -DPARQUET_BUILD_EXECUTABLES=ON
-      -GNinja
     ]
-
     args << "-DARROW_MIMALLOC=ON" unless Hardware::CPU.arm?
+    args << "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-dead_strip_dylibs" if OS.mac? # to reduce overlinking
+    if build.bottle? && Hardware::CPU.intel? && (!OS.mac? || !MacOS.version.requires_sse42?)
+      args << "-DARROW_SIMD_LEVEL=NONE"
```

Comment:
This is meant to do runtime dispatch, so IIRC upstream didn't want us to disable SIMD here.

Reply:
The default minimum is SSE4.2.

According to https://github.com/apache/arrow/blob/apache-arrow-18.1.0/docs/source/cpp/env_vars.rst

> For example, by default, SSE4.2 optimizations are enabled on x86
> builds: therefore, **with this default setting, Arrow C++ does not work at**
> **all on a CPU without support for SSE4.2.**  This setting can be changed
> using the ``ARROW_SIMD_LEVEL`` CMake variable so as to either raise or
> lower the optimization level.